### DISCUSSION TITLE
检测mingw的版本以及线程库

### DISCUSSION THREAD

**Discussion by jingkaimori** [xmake-io/xmake#4044] [Ideas]
目前无法[在脚本中获取mingw的版本](https://github.com/xmake-io/xmake/discussions/3782#discussioncomment-6587596)，也不知道当前的mingw工具链的`libstdc++`使用的是win32线程API还是posix线程API。使用win32线程API的mingw没有`std::thread`相关的定义，在doctest中会引起编译错误。

* https://stackoverflow.com/questions/17242516/mingw-w64-threads-posix-vs-win32
* https://github.com/doctest/doctest/issues/696#issuecomment-1250116107

**Comment by waruqi** [comment]
可以自己检测内置宏，https://github.com/xmake-io/xmake/blob/0ffb589fb816c4d886cf88d322df45f7ebe5434e/tests/apis/check_xxx/xmake.lua#L49

也可以自己 dump 一些编译器信息。。https://github.com/xmake-io/xmake/blob/0ffb589fb816c4d886cf88d322df45f7ebe5434e/tests/apis/check_xxx/xmake.lua#L27
